File: Samples/iOS/Objective-C/IOS-SDK-Demo-App/Ad Formats/LOInterScrollerViewController.m
//
//  LOInterScrollerViewController.m
//  IOS-SDK-Demo-App
//
//  Created by Manoj Budumuru on 10/8/24.
//

#import "LOInterScrollerViewController.h"

@interface LOInterScrollerViewController () <VungleBannerViewDelegate>
@property (nonatomic, strong) VungleBannerView *banner;
@end

@implementation LOInterScrollerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = @"Liftoff InterScroller display";
    // Do any additional setup after loading the view.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    CGRect rect = self.view.safeAreaLayoutGuide.layoutFrame;
    self.heightConstraint.constant = rect.size.height;
    [self.view layoutIfNeeded];
    self.banner = [[VungleBannerView alloc] initWithPlacementId:self.placementId
                                                                vungleAdSize:[VungleAdSize VungleAdSizeFromCGSize:rect.size]];
    self.banner.delegate = self;
    [self.banner load:nil];
    [self.adView addSubview:self.banner];
}

- (void)setAdLayoutConstraints {
    if (self.banner != nil) {
        self.banner.translatesAutoresizingMaskIntoConstraints = false;
        NSArray<NSLayoutConstraint *> *constraintArray = @[[self.banner.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor],
                                                           [self.banner.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor],
                                                           [self.banner.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
                                                           [self.banner.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor],
                                                           [self.banner.heightAnchor constraintEqualToConstant:self.view.safeAreaLayoutGuide.layoutFrame.size.height]];
        [NSLayoutConstraint activateConstraints:constraintArray];
    }
}

- (void)bannerAdDidLoad:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdDidLoad");
}

- (void)bannerAdWillPresent:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdWillPresent");
}

- (void)bannerAdDidPresent:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdDidPresent");
    [self setAdLayoutConstraints];
}

- (void)bannerAdDidFail:(VungleBannerView * _Nonnull)bannerView withError:(NSError * _Nonnull)withError {
    NSLog(@"bannerAdDidFail");
}

- (void)bannerAdWillClose:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdWillClose");
}

- (void)bannerAdDidClose:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdDidClose");
}

- (void)bannerAdDidTrackImpression:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdDidTrackImpression");
}

- (void)bannerAdDidClick:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdDidClick");
}

- (void)bannerAdWillLeaveApplication:(VungleBannerView * _Nonnull)bannerView {
    NSLog(@"bannerAdWillLeaveApplication");
}

@end

File: Samples/iOS/Swift/IOS-SDK-Demo-App/Ad Formats/LOInterScrollerViewController.swift
//
//  LOInterScrollerViewController.swift
//  IOS-SDK-Demo-App
//
//  Created by Manoj Budumuru on 10/4/24.
//

import UIKit
import VungleAdsSDK

class LOInterScrollerViewController: UIViewController {
    var placementId: String?
    var banner: VungleBannerView?
    
    @IBOutlet weak var adView: UIView!
    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "Liftoff InterScroller display"
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        let rect: CGRect = view.safeAreaLayoutGuide.layoutFrame
        heightConstraint.constant = rect.height
        view.layoutIfNeeded()
        if let placementId = placementId {
            let bannerAd = VungleBannerView(placementId: placementId, vungleAdSize: .VungleAdSizeFromCGSize(rect.size))
            bannerAd.delegate = self
            bannerAd.load()
            adView.addSubview(bannerAd)
            banner = bannerAd
        }
    }
    
    // use this method to set the constarints to the Ad.
    func setAdLayoutConstraints() {
        if let bannerad = banner {
            bannerad.translatesAutoresizingMaskIntoConstraints = false
            NSLayoutConstraint.activate([
                bannerad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
                bannerad.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
                bannerad.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
                bannerad.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
                bannerad.heightAnchor.constraint(equalToConstant: view.safeAreaLayoutGuide.layoutFrame.height)
            ])
        }
    }
}

extension LOInterScrollerViewController: VungleBannerViewDelegate {
    func bannerAdDidLoad(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdDidLoad")
    }

    func bannerAdWillPresent(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdWillPresent")
    }

    func bannerAdDidPresent(_ bannerView: VungleAdsSDK.VungleBannerView) {
        self.setAdLayoutConstraints()
        print("bannerAdDidPresent")
    }

    func bannerAdDidFail(_ bannerView: VungleAdsSDK.VungleBannerView, withError: NSError) {
        let alert = UIAlertController(title: "Load Ad Error", message: withError.localizedDescription, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "Dismiss", style: .default))
        present(alert, animated: true,completion: nil)
    }

    func bannerAdWillClose(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdWillClose")
    }

    func bannerAdDidClose(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdDidClose")
    }

    func bannerAdDidTrackImpression(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdDidTrackImpression")
    }

    func bannerAdDidClick(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdDidClick")
    }

    func bannerAdWillLeaveApplication(_ bannerView: VungleAdsSDK.VungleBannerView) {
        print("bannerAdWillLeaveApplication")
    }
}
